src/opentaskpy/taskhandlers/batch.py

```diff
@@ -346,16 +365,27 @@ def run(self, kill_event: threading.Event | None = None) -> bool:
                     logged = True
 
                 batch_task["logged_status"] = logged
+                batch_task["logged_status"] = logged
+                batch_task["logged_status"] = logged
 
             # Check if there are any tasks that are still in RUNNING state, if not then we are done
-            remaining_tasks = [
+            running_tasks = [
                 task
                 for task in self.task_order_tree.values()
-                if task["status"] == "RUNNING" or task["status"] == "NOT_STARTED"
+                if task["status"] in ["RUNNING"]
             ]
-            self.logger.debug(f"Remaining tasks: {remaining_tasks}")
-            if len(remaining_tasks) == 0:
-                self.logger.debug("No remaining tasks, breaking out of loop")
+            self.logger.debug(f"Running tasks: {running_tasks}")
+            if len(running_tasks) == 0:
```

Comment:
@adammcdonagh 
I'm wandering whether there may be a delay between Running/Complete tasks and Not_Started actually moving into Running state. What do you think? Is there a chance of not running specific tasks with this in place?

Reply:
This is what I'm thinking it might be the case

`> if task["status"] == "RUNNING" or task["status"] == "NOT_STARTED"`

Reply:
We will have to see how this behaves. I cannot replicate this directly.